From dba6abf5010b6d63ef745e08b334ee3f1361424f Mon Sep 17 00:00:00 2001
From: Andre McCurdy <armccurdy@gmail.com>
Date: Thu, 12 Nov 2015 15:38:37 -0800
Subject: [PATCH] rockford/middleware/v3d/driver/interface/vcos/vcos_types.h:
 gcc5 inline fixes

gcc5 defaults to C99 inline semantics, which means "extern inline"
functions are treated differently compared to previous versions of gcc.

  https://gcc.gnu.org/gcc-5/porting_to.html

Adding the function attribute "gnu_inline" forces the old GNU89 inline
semantics to be used.

Signed-off-by: Andre McCurdy <armccurdy@gmail.com>
---
 rockford/middleware/v3d/driver/interface/vcos/vcos_types.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/rockford/middleware/v3d/driver/interface/vcos/vcos_types.h b/rockford/middleware/v3d/driver/interface/vcos/vcos_types.h
index bcab6fb..9486e7c 100644
--- a/rockford/middleware/v3d/driver/interface/vcos/vcos_types.h
+++ b/rockford/middleware/v3d/driver/interface/vcos/vcos_types.h
@@ -110,8 +110,8 @@ typedef enum
 #if defined(NDEBUG)
 
 #ifdef __GNUC__
-# define VCOS_INLINE_DECL extern __inline__
-# define VCOS_INLINE_IMPL static __inline__
+# define VCOS_INLINE_DECL extern __inline__ __attribute__((gnu_inline))
+# define VCOS_INLINE_IMPL static __inline__ __attribute__((gnu_inline))
 #else
 # define VCOS_INLINE_DECL static _VCOS_INLINE   /* declare a func */
 # define VCOS_INLINE_IMPL static _VCOS_INLINE   /* implement a func inline */
-- 
1.9.1

